src/parse.cpp

```diff
@@ -1,10 +1,50 @@
 #include "parse.hpp"
-
 #include "lexer.hpp"
+#include <iostream>
+#include <stdexcept>
+
+std::unique_ptr<FunctionAST> parseFunction(TokenStream &stream);
+std::vector<std::string> parseVarInput(TokenStream &stream);
 
-// TODO: This return type might need to change
-std::unique_ptr<FunctionAST> parse(std::vector<Token> tokens)
+std::unique_ptr<FunctionAST> parse(TokenStream &stream)
 {
-    // TODO: Implement parsing
+    auto tok = stream.next();
+    if (tok.getkey() == "KEYWORD" && tok.getvalue() == "FUNCTION")
+    {
+        std::cout << "Parsing function..." << std::endl;
+        parseFunction(stream);
+    }
     return nullptr;
```

Comment:
Instead of always returning `nullptr`, this function should throw with an error like "Expected keyword FUNCTION, but got  blah blah"

Reply:
This has been corrected will be committed shortly